src/services/State.ts

```diff
@@ -76,24 +77,21 @@ export class StateService {
     return { first, last };
   }
 
-  setSelected(selected: any[]): StateService {
-    if (!this.selected) {
-      this.selected = selected || [];
-    } else {
-      this.selected.splice(0, this.selected.length);
-      this.selected.push(...selected);
-    }
+  get selected(): any[] {
+    return this.rows.filter(row => this.isRowSelected(row));
```

Comment:
Could `get `selected()` and `setSelected(selected: any[])` be merged as the get/set property